app/components/data/action/add/file-group-create-form.component.ts

```diff
@@ -0,0 +1,152 @@
+
+import { NotificationService } from "app/components/base/notifications";
+import { SidebarRef } from "app/components/base/sidebar";
+import { DynamicForm } from "app/core";
+import { BlobContainer, ServerError } from "app/models";
+import { FileGroupCreateDto } from "app/models/dtos";
+import { CreateFileGroupModel, createFileGroupFormToJsonData, fileGroupToFormModel } from "app/models/forms";
+import { AccountService, PythonRpcService, StorageService } from "app/services";
+import { AdalService } from "app/services/adal";
+import { Constants, log } from "app/utils";
+import * as path from "path";
+
+import "./file-group-create-form.scss";
+
+@Component({
+    selector: "bl-file-group-create-form",
+    templateUrl: "file-group-create-form.html",
+})
+export class FileGroupCreateFormComponent extends DynamicForm<BlobContainer, FileGroupCreateDto> implements OnInit {
+    public folder: string;
+    public groupExists: boolean;
+    public editing: boolean;
+    public title: string = "Create file group";
+    public description: string = "Upload files into a managed storage container that you can use " +
+        "for resource files in your jobs and tasks";
+
+    constructor(
+        public sidebarRef: SidebarRef<FileGroupCreateFormComponent>,
+        private adalService: AdalService,
+        private accountService: AccountService,
+        private formBuilder: FormBuilder,
+        private notificationService: NotificationService,
+        private pythonRpcService: PythonRpcService,
+        private storageService: StorageService) {
+        super(FileGroupCreateDto);
+
+        const validation = Constants.forms.validation;
+        this.form = this.formBuilder.group({
+            name: ["", [
+                Validators.required,
+                Validators.maxLength(validation.maxLength.fileGroup),
+                Validators.pattern(validation.regex.fileGroup),
+            ], [
+                this._validateFileGroupName.bind(this),
+            ]],
+            folder: ["", [Validators.required]],
+            includeSubDirectories: [true],
+            options: [null, []],
+            accessPolicy: ["private"],
+        });
+    }
+
+    public ngOnInit() {
+        this.editing = false;
+    }
+
+    @autobind()
+    public submit(): Observable<any> {
+        const formData = this.getCurrentValue();
+        const batch = Constants.ResourceUrl.batch;
+        const arm = Constants.ResourceUrl.arm;
+
+        // We now need to pass in both the Batch and ARM tokens as we are dealing with storage.
+        const observable = this.accountService.currentAccount.cascade((account) => {
+            return this.adalService.accessTokenFor(account.subscription.tenantId, batch).cascade((batchToken) => {
+                return this.adalService.accessTokenFor(account.subscription.tenantId, arm).cascade((armToken) => {
+                    return this.pythonRpcService.call("create_file_group", [
+                        batchToken,
```

Comment:
I would move all of this nested thing in a file-group service or ncj something. With maybe another method in callPython that also include the token as first param.